doc/source/acb_theta.rst

```diff
@@ -0,0 +1,1540 @@
+
+    Assuming that *mat* is a symplectic matrix of size `2r\times 2r` and *res*
+    is square of size `2g\times 2g` for some `g\geq r`, sets *res* to the symplectic matrix
+
+        .. math ::
+
+            \begin{pmatrix} \alpha && \beta & \\ & I_{g-r} && 0_{g-r} \\ \gamma &&\delta &\\ & 0_{g-r} && I_{g-r} \end{pmatrix}
+
+    where `\alpha,\beta,\gamma,\delta` are the `r\times r` blocks of *mat*.
+
+.. function:: void sp2gz_restrict(fmpz_mat_t res, const fmpz_mat_t mat)
+
+    Assuming that *mat* is a symplectic matrix of size `2g\times 2g` and *res*
+    is square of size `2r\times 2r` for some `r\leq g`, sets *res* to the
+    matrix whose `r\times r` blocks are the upper left corners of the
+    corresponding `g\times g` block of *mat*. The result may not be a
+    symplectic matrix.
+
+.. function:: slong sp2gz_nb_fundamental(slong g)
+
+    Returns the number of fundamental symplectic matrices used in the reduction
+    algorithm on `\mathbb{H}_g`. This number is 1 when `g=1` (the `J` matrix)
+    and 19 when `g=2` [Got1959]_. When `g>2`, a complete set of matrices
+    defining the boundary of a fundamental domain for the action of
+    `\mathrm{Sp}_{2g}(\mathbb{Z})` is not currently known. As a substitute, we
+    consider two types of matrices: the `19 g(g-1)/2` matrices obtained by
+    mimicking the `g=2` matrices on any pair of indices between 0 and `g-1`,
+    and the `2^g` matrices obtained by embedding a copy of a lower-dimensional
+    `J` matrix on any subset of indices.
+
+.. function:: void sp2gz_fundamental(fmpz_mat_t mat, slong j)
+
+    Sets *mat* to the `j^{\text{th}}` fundamental symplectic matrix as defined
+    above.
+
+.. function:: int sp2gz_is_correct(const fmpz_mat_t mat)
+
+    Returns true (nonzero) iff *mat* is a symplectic matrix.
+
+.. function:: int sp2gz_is_j(const fmpz_mat_t mat)
+
+    Returns true (nonzero) iff the symplectic matrix *mat* is the `J` matrix.
+
+.. function:: int sp2gz_is_block_diag(const fmpz_mat_t mat)
+
+    Returns true (nonzero) iff the symplectic matrix *mat* is of block-diagonal
+    form as in :func:`sp2gz_block_diag`.
+
+.. function:: int sp2gz_is_trig(const fmpz_mat_t mat)
+
+    Returns true (nonzero) iff the sympletic matrix *mat* is of trigonal form
+    as in :func:`sp2gz_trig`.
+
+.. function:: int sp2gz_is_embedded(fmpz_mat_t res, const fmpz_mat_t mat)
+
+    Assuming that *mat* is a `2g\times 2g` symplectic matrix and *res* is
+    square of size `2r` for some `r\leq g`, returns true (nonzero) iff *mat*
+    can be obtained as the result of :func:`sp2gz_embed` from a `2r\times 2r`
+    symplectic matrix, and store this matrix in *res*. Otherwise, returns
+    false (0) and leaves *res* undefined.
+
+.. function:: void sp2gz_inv(fmpz_mat_t inv, const fmpz_mat_t mat)
+
+    Sets *inv* to the inverse of the symplectic matrix *mat*.
+
+.. function:: fmpz_mat_struct* sp2gz_decompose(slong* nb, const fmpz_mat_t mat)
```

Comment:
We typeset pointers in FLINT as `type * var`. I saw a lot of other occurrences of this, both in the documentation and in the code.